### ISSUE TITLE
Blown-off Weapon Use  Allowed

### ISSUE THREAD

**Issue by BairdEC** [MegaMek/megamek#1042]
### Environment
MegaMek version 0.44.0 (w/MekHQ)
Win10
Java Version 1.8.0_171

### Description
MM is allowing the use of  weapons in arms that were blown off by torso destruction in previous scenarios.

### Files

![invalid weapon](https://user-images.githubusercontent.com/38081112/42606664-c5f589a4-8533-11e8-9461-aefa766c7729.png)

[autosave.sav.gz](https://github.com/MegaMek/megamek/files/2186672/autosave.sav.gz) (Metal Magic)

**Comment by BairdEC** [comment]
Yes, I was.  I had to reload the save to get the log message, though.

Weapons fire for Panther PNT-9R (Metal Magic)
    PPC at Locust LCT-1V (Federated Suns); needs 11, rolls 10 : misses

**Comment by BLOODWOLF333** [comment]
![github](https://user-images.githubusercontent.com/16950832/42612105-04fc6a3e-855f-11e8-90df-427fa1b75474.png)

Verified, this might be specific to HQ though.  His right side armor and IS are gone but under the systems tab it shows all equipment in the RA as fine.

[1042FirePhase.sav.gz](https://github.com/MegaMek/megamek/files/2187020/1042FirePhase.sav.gz)

**Comment by BairdEC** [comment]
FWIW, the MekHQ hanger shows no damage to the arm.  It was detached only because the torso was destroyed.


  | Internal | Armor |   |  
-- | -- | -- | -- | --
Head | 3 | 9 |   |  
Center Torso | 11 | 8 |   |  
Center Torso (rear) |   | 7 |   |  
Right Torso | X | X |   |  
Right Torso (rear) |   | X |   |  
Left Torso | 8 | 10 |   |  
Left Torso (rear) |   | 5 |   |  
Right Arm | 6 | 10 |   | BLOWN OFF
Left Arm | 6 | 6 |   |  
Right Leg | 8 | 10 |   |  
Left Leg | 8 | 12 |  

[Metal Magic29970619.zip](https://github.com/MegaMek/megamek/files/2187034/Metal.Magic29970619.zip)

**Comment by arlith** [comment]
This is a MHQ problem.  It happens in `ResolveScenario.checkForLostLimbs`.  If the player controls the  battlefield, all of the missing status is cleared from the equipment.  This doesn't seem quite right to me, but I'm not super familiar with MHQ.

`Entity.destroyLocation(2)` handles blowing off locations, and just marks equipment in blown-off locations as missing.

**Comment by arlith** [comment]
This is a MHQ problem.  It happens in `ResolveScenario.checkForLostLimbs`.  If the player controls the  battlefield, all of the missing status is cleared from the equipment.  This doesn't seem quite right to me, but I'm not super familiar with MHQ.

`Entity.destroyLocation(2)` handles blowing off locations, and just marks equipment in blown-off locations as missing.

**Comment by SJuliez** [comment]
Forwarding this to MHQ as per Arlith's comment. Note that it says "torso destruction in previous scenarios".
